src/createStore.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-console */
+import _ from 'lodash';
+import { BehaviorSubject } from 'rxjs';
+
+class BaseStore {
+  constructor(options = {}) {
+    this.options = {
+      initialState: null,
+      thunkArgument: null,
+      appendAction: false,
+      reducer: state => state,
+      enableLogger: true,
+      console: console,
+      ...options,
+    };
+
+    this.internalState$ = new BehaviorSubject(this.options.initialState)
+      .scan((previousState, action) => {
+        let updatedState;
+        const d = new Date();
+        const prettyDate = `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}.${d.getMilliseconds()}`;
+
+        try {
+          updatedState = this.options.reducer(previousState, action);
+        } catch (error) {
+          if (action && action.type) {
+            this.options.console.error(`Error processing @ ${prettyDate} ${action.type}:`);
+          }
+          this.options.console.error(error);
+
+          return previousState;
+        }
+
+        // logger in non-production mode only
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
Perhaps this logic should go to the logger itself, or better yet, whenever you create an instance of the logger you _could_ specify the environment(s) in which it should work?

Why? Because of this: https://github.com/nodejs/node/issues/3104

Another way would be to cache that value at the beginning of the module. :)